test/services/SdesGatekeeperServiceSpec.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.*
+import models.FileRole.CashStatement
+import models.metadata.{CashStatementFileMetadata, Metadata, MetadataItem}
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.{times, verify, when}
+import play.api.Application
+import play.api.i18n.Messages
+import utils.SpecBase
+
+class SdesGatekeeperServiceSpec extends SpecBase {
+
+  "SdesGatekeeperService" should {
+
+    "convert FileInformation to CashStatementFile correctly" in new Setup {
+      val sdesFileInformation: FileInformation = validCashStatementFileInformation
+
+      val result: CashStatementFile = sdesGatekeeperService.convertToCashStatementFile(sdesFileInformation)(msg)
+
+      result.filename mustBe sdesFileInformation.filename
+      result.downloadURL mustBe sdesFileInformation.downloadURL
+      result.metadata.periodStartYear mustBe periodStartYear
+      result.metadata.periodStartMonth mustBe periodStartMonth
+      result.metadata.periodStartDay mustBe periodStartDay
+      result.metadata.periodEndYear mustBe periodEndYear
+      result.metadata.periodEndMonth mustBe periodEndMonth
+      result.metadata.periodEndDay mustBe periodEndDay
+      result.metadata.fileFormat mustBe FileFormat(csv)
+      result.metadata.fileRole mustBe CashStatement
+      result.metadata.statementRequestId mustBe someRequestId
+    }
+  }
+
+  "SdesGatekeeperService.convertTo" should {
```

Comment:
There should be no need to use mock here. Should have been done by importing the relevant implicits. Will be taken care later.